sw/device/lib/crypto/drivers/kmac.h

```diff
@@ -87,8 +166,8 @@ kmac_error_t kmac_init(kmac_operation_t operation_type);
  * from the final Keccak state.
  */
 OT_WARN_UNUSED_RESULT
-kmac_error_t kmac_update(const uint8_t *data, size_t data_len, uint8_t *digest,
-                         size_t digest_len);
+kmac_error_t kmac_update(kmac_operation_t operation, const uint8_t *data,
+                         size_t data_len, uint8_t *digest, size_t digest_len);
```

Comment:
I think rather than having a one-shot `kmac_init()` and `kmac_update()`, it might make more sense to expose a single one-shot `kmac()` function (or even clearer, one `kmac()`, one `cshake()`, one `shake()`, and one `sha3()`). The name `update` makes me think I can call it multiple times (i.e. streaming), and if I can't then it just seems to make sense to collapse `init` and `update` into a true one-shot.

Reply:
In hindsight, I think it makes more sense rename `kmac_update` to `kmac_process_msg_blocks` or something similar (which I will do). This function is used by each of {SHA3, SHAKE, CSHAKE, KMAC} as a common routine for feeding message blocks. 

As for combining these sequence of functions call under one-shot `kmac()`, `shake()` etc. I was thinking a follow-up PR would be better. I still need to do additional work to connect these driver calls to higher level API, and there this would be one of TODO items on the list. WDYT?

Reply:
Yep, fine with me to do this as a follow-up PR.